src/library/scala/collection/mutable/GrowableBuilder.scala

```diff
@@ -25,7 +25,7 @@ package collection.mutable
   * @define Coll `GrowingBuilder`
   * @define coll growing builder
   */
-class GrowableBuilder[Elem, To <: Growable[Elem]](protected val elems: To)
+class GrowableBuilder[Elem, To <: Growable[Elem] with IterableOnce[Elem]](protected val elems: To)
```

Comment:
it's not clear to me why `To` should have to extend `IterableOnce`. I understand the desire to have a `knownSize` implementation on `elems` to call, but it seems arbitrary to that it's a bound on `GrowableBuilder#To` when it's not a bound on `Growable` or `Builder#To`.

I kind of feel like it would be better to define `knownSize` (still defaulting to `-1`) on `Growable`, since:

- there's no reason to assume it *can't* know its size
- it could theoretically be helpful for it to be able to get its size if it is known
- it doesn't seem to me to increase the maintenance burden—you don't ever *have* to override it in `Growable` anyway, and it's just moving where it's defined

Reply:
I suppose you're right. Growables are either Builders or full collections.